test_conformance/workgroups/test_wg_suggested_local_work_size.cpp

```diff
@@ -0,0 +1,626 @@
+    if (dyn_mem_size)
+    {
+        err = clSetKernelArg(scan_kernel, 1, dyn_mem_size, NULL);
+        test_error_ret(err, "clSetKernelArg failed", -1);
+    }
+    err =
+        clEnqueueNDRangeKernel(queue, scan_kernel, work_dim, global_work_offset,
+                               test_values, // global work size
+                               NULL, 0, NULL, NULL);
+    test_error_ret(err, "clEnqueueNDRangeKernel failed", -1);
+    err = clEnqueueReadBuffer(queue, buffer, CL_NON_BLOCKING, 0,
+                              sizeof(kernel_work_size), kernel_work_size, 0,
+                              NULL, NULL);
+    test_error_ret(err, "clEnqueueReadBuffer failed", -1);
+    err = clFinish(queue);
+    test_error_ret(err, "clFinish failed", -1);
+
+    if (kernel_work_size[0] != local_work_size[0]
+        || kernel_work_size[1] != local_work_size[1]
+        || kernel_work_size[2] != local_work_size[2])
+    {
+        std::cout
+            << "Kernel work size differs from local work size suggested:\n"
+            << "Kernel work size: (" << kernel_work_size[0] << ", "
+            << kernel_work_size[1] << ", " << kernel_work_size[2] << ")"
+            << "Local work size: (" << local_work_size[0] << ", "
+            << local_work_size[1] << ", " << local_work_size[2] << ")\n";
+        return -1;
+    }
+    return err;
+}
+
+int do_test_work_group_suggested_local_size(
+    cl_device_id device, cl_context context, cl_command_queue queue,
+    bool (*skip_cond)(size_t), size_t start, size_t end, size_t incr,
+    cl_long max_local_mem_size, size_t global_work_offset[], num_dims dim)
+{
+    clProgramWrapper scan_program;
+    clKernelWrapper scan_kernel;
+    clProgramWrapper scan1kb_program;
+    clKernelWrapper scan1kb_kernel;
+    clProgramWrapper scan4kb_program;
+    clKernelWrapper scan4kb_kernel;
+    clProgramWrapper scan16kb_program;
+    clKernelWrapper scan16kb_kernel;
+    clProgramWrapper scan32kb_program;
+    clKernelWrapper scan32kb_kernel;
+    clProgramWrapper scan_dyn_program;
+    clKernelWrapper scan_dyn_kernel;
+    int err;
+    size_t test_values[] = { 1, 1, 1 };
+
+    // Create the kernels needed
+    err = create_single_kernel_helper(
+        context, &scan_program, &scan_kernel, 1, &wg_scan_local_work_group_size,
+        "test_wg_scan_local_work_group_size", "-DLOCAL_MEM_SIZE=1");
+    test_error_ret(err, "create_single_kernel_helper failed", -1);
+    err = create_single_kernel_helper(
+        context, &scan1kb_program, &scan1kb_kernel, 1,
+        &wg_scan_local_work_group_size,
+        "test_wg_scan_local_work_group_size_static_local",
+        "-DLOCAL_MEM_SIZE=1024");
+    test_error_ret(err, "create_single_kernel_helper failed", -1);
+    err = create_single_kernel_helper(
+        context, &scan4kb_program, &scan4kb_kernel, 1,
+        &wg_scan_local_work_group_size,
+        "test_wg_scan_local_work_group_size_static_local",
+        "-DLOCAL_MEM_SIZE=4096");
```

Comment:
Embedded profile devices are only required to support 1KB local memory.

Elsewhere in the test we're already gating test execution based on the value of  `CL_DEVICE_LOCAL_MEM_SIZE`. Please can we also ensure that programs aren't built unless they aren't going to request more than the amount of local memory supported by the device.